debug_module.tex

```diff
@@ -56,25 +56,46 @@ \section{Debug Module Interface (DMI)} \label{dmi}
 on this DMI, the base address of the next DM in the DMI address space is given
 in \RdmNextdm.
 
-
 The Debug Module is controlled via register accesses to its DMI address space.
 
 \section{Reset Control} \label{reset}
 
-The Debug Module controls a global reset signal, \FdmDmcontrolNdmreset
-(non-debug module reset),
-which can reset, or hold in reset, every component in the platform,
-except for the Debug Module and Debug Transport Modules.
-Exactly what is affected by this reset is implementation dependent, as long as
-it is possible to debug programs from the first instruction executed.
+There are two methods that allow a debugger to reset harts.
+\FdmDmcontrolNdmreset resets all the harts in the system, as well as all other
+parts of the system except for the Debug Module and Debug Transport Modules.
+Exactly what is affected by this reset is implementation dependent, but it
+must be possible to debug programs from the first instruction executed.
+\FdmDmcontrolHartreset resets all the currently selected harts. In this case an
+implementation may reset more harts than just the ones that are selected. The
+debugger can discover which other harts are reset (if any) by selecting them
+and checking \FdmDmstatusAnyhavereset and \FdmDmstatusAllhavereset.
+
+To perform either of these resets, the debugger first asserts the bit, and then
+clears it. The actual reset may start as soon as the bit is asserted, but is allowed
+to start a long time after the bit is deasserted. While the reset is
+on-going, harts are either in the running state, indicating it's possible to
+perform some abstract commands during this time, or in the unavailable state,
+indicating it's not possible to perform any abstract commands during this time.
+Once a hart's reset is complete, {\tt havereset} becomes set.  When a hart comes out
+of reset and \FdmDmcontrolHaltreq or \Fresethaltreq are set, the hart will
+immediately enter Debug Mode. Otherwise it will execute normally.
```

Comment:
```suggestion
immediately enter Debug Mode and be in the halted state. Otherwise it will execute normally and be in the running state.
```

Reply:
This implies that being in Debug Mode doesn't automatically mean being halted, but being in Debug Mode is synonymous with being halted. If that's not clear, we could use something like:
immediately enter Debug Mode (be in the halted state). Otherwise it will execute normally (be in the running state).

What do you think?

Reply:
That sounds good.  It's clear to me that they're synonymous but I just thought that because the rest of this section talks about the various states, they should be mentioned in this sentence, too.